### ISSUE TITLE
Status des Registers 13011 Load 1 ON/Off im Intelligenten Modus

### ISSUE THREAD

**Issue by tttooommmeeekkk** [dl1ekm/sungrow_do_port#1]
Tolle Arbeit - sehr schöne Erweiterung. Danke dafür! 

Das Register 13011scheint offenbar im Intelligenen Modus nicht den Status des Relais auszugeben.

Gibt es hier vielleicht noch ein weiteres ( vielleicht ein nicht dokumentiertes read-only?)-Register, dass den Schaltzustand des DO-Relais im Intelligenten Modus (Schalten bei PV-Überschuss, sofern im Zeitfenster) bzw. in egal welchem DO-Konfigurations-Modus anzeigt anzeigt?
Da ich die Sungrow-App (iSolarCloud) nicht nutze, kann ich leider nicht dort nachschauen, wie es dort gelöst/angezeigt wird.

By the way: Das Register 13011 lässt sich auch trotz ausgewähltem Intelligenten Modus schalten (hab es per WR-Webinterface jedenfalls schalten können).
So ganz verstehe ich daher die Logik von Sungrow hier noch nicht. Ich hätte erwartet, dass das Register 13011 auch im Intelligenten Modus geschaltet/geändert wird. Das ist leider auch im "Sungrow Communication Protocol V1.1.2" nicht erklärt.

Viele Grüße
Tommi

**Comment by elektrinis** [comment]
Have you figured this out? I need to switch DO port on/off when I want, and see the status.

**Comment by tttooommmeeekkk** [comment]
No, obviously there is no such register by using the "intelligent mode".

But in "on/off-mode" and switching the regsiter 13011 the state of the register reads back the actual value.
So I meanwhile only use the "switch-mode" for the DO-relay and wrote my own "intelligent mode" in homeassistant based on that switching-mode - so that the actual sensor-status of the DO-relay is visible.

**Comment by wenegi88** [comment]
> Tolle Arbeit - sehr schöne Erweiterung. Danke dafür!
> 
> Das Register 13011scheint offenbar im Intelligenen Modus nicht den Status des Relais auszugeben.
> 
> Gibt es hier vielleicht noch ein weiteres ( vielleicht ein nicht dokumentiertes read-only?)-Register, dass den Schaltzustand des DO-Relais im Intelligenten Modus (Schalten bei PV-Überschuss, sofern im Zeitfenster) bzw. in egal welchem DO-Konfigurations-Modus anzeigt anzeigt? Da ich die Sungrow-App (iSolarCloud) nicht nutze, kann ich leider nicht dort nachschauen, wie es dort gelöst/angezeigt wird.
> 
> By the way: Das Register 13011 lässt sich auch trotz ausgewähltem Intelligenten Modus schalten (hab es per WR-Webinterface jedenfalls schalten können). So ganz verstehe ich daher die Logik von Sungrow hier noch nicht. Ich hätte erwartet, dass das Register 13011 auch im Intelligenten Modus geschaltet/geändert wird. Das ist leider auch im "Sungrow Communication Protocol V1.1.2" nicht erklärt.
> 
> Viele Grüße Tommi

Wie hast du das in der .yaml gelöst? und hast du das in der dashboard .yaml auch abgebildet?

**Comment by tttooommmeeekkk** [comment]
> Wie hast du das in der .yaml gelöst? und hast du das in der dashboard .yaml auch abgebildet?

Ich bin so vorgegangen, wie hier beschrieben (ich fand das sehr gut*):
https://github.com/dl1ekm/sungrow_do_port
Habe mir also auch ein neues Dashboard in HA angelegt und den Code aus der hier im Projekt abgelegten dashboard.yaml eingefügt bzw. diesen Code in eines meiner bestehenden Dashboards eingearbeitet.

*hab allerdings eine Änderung vorgenommen, wegen einer unpassenden Faktorisierung  (siehe meinen anderen Issue in diesem Projekt).
Auch ein paar Zeilen musste ich im Code auskommentieren, da meine HA-Version damit Probleme hatte (Näheres dazu z.B. [hier](https://www.photovoltaikforum.com/thread/166134-daten-lesen-vom-sungrow-wechselrichtern-modbus/?postID=3690330#post3690330) )

**Comment by wenegi88** [comment]
Ja super, danke dir! Bei mir funktioniert es nun auch Problemlos. Die Rückmeldung vom geschalteten DO hast du auch nirgends gefunden oder?! Ich meine PV-GO hat es mit einem zusätzlichen Shelly gelöst....

**Comment by tttooommmeeekkk** [comment]
Die DO-Zustands-Änderung (Register 13011) wird leider nur im "Schaltermodus" geschrieben. Im "intelligenten Modus" wird dieses Register leider nicht geschrieben (und man sieht dort nicht, wie der Zustand ist).
Ich habe das bei mir so gelöst, dass ich den "Schaltermodus" nutze und die "Intelligenz" im HA nachgebaut habe (Vorteil: man ist viel flexibler bei der Wahl der Bedingungen...). Dann hat man das gewünschte bedingte Schalten und die Zustandsänderung des Schalters ist aktiv sichtbar.

So hab ich die "Intelligenz" als Automation bspw. bei mir umgesetzt:

**DO EIN**
alias: DO-Schalter Warmwasser EIN
description: >-
  stellt das DO-Relay des WR auf "geschlossen" (EIN) und schaltet damit die
  Warmwasser-Schnell-Funktion der Wärmepumpe ein; sofern Uhrzeit zwischen 8-18
  Uhr und Netzeinspeisung größer 1.900W und das Relay für mehr, als 1 Stunde AUS
  war
trigger:
  - platform: state
    entity_id:
      - sensor.export_power
    for:
      hours: 0
      minutes: 0
      seconds: 10
condition:
  - condition: time
    after: "08:00:00"
    before: "18:00:00"
  - condition: numeric_state
    entity_id: sensor.export_power
    above: 1900
  - condition: state
    entity_id: input_select.set_sg_do_switch_mode
    state: Aus
    for:
      hours: 1
      minutes: 0
      seconds: 0
action:
  - service: input_select.select_option
    data:
      option: Ein
    target:
      entity_id: input_select.set_sg_do_switch_mode
mode: single

**DO AUS**
alias: DO-Schalter Warmwasser AUS
description: >-
  stellt das DO-Relay des WR auf "offen" und schaltet damit die
  Warmwasser-Schnell-Funktion der Wärmepumpe aus; sofern Netzeinspeisung kleiner
  1.100W für länger, als 2h
trigger:
  - platform: numeric_state
    entity_id:
      - sensor.export_power
    for:
      hours: 2
      minutes: 0
      seconds: 0
    below: 1100
condition: []
action:
  - service: input_select.select_option
    data:
      option: Aus
    target:
      entity_id: input_select.set_sg_do_switch_mode
mode: single

**Comment by wenegi88** [comment]
Hört sich gut an, danke vielmals, ich erweitere das Procedere noch mit der Abfrage der Batterie, damit der Heizstab nicht diese zu weit runter zieht
